### DISCUSSION TITLE
encode/databases best practices

### DISCUSSION THREAD

**Discussion by Amiram1** [fastapi/fastapi#2628] [Questions] [answered]
Hi guys,

Trying to figure out the best workflow for async db using databases, I have two questions:

1) when using startup events - 
```
@app.on_event("startup")
async def startup():
    await database.connect()
```
How do I then pass the connection to different routers? When using sqlalchemy the suggestion is to use a db session generator using dependency injection, what is the similar flow using databases?

2) Is there a way to work with two different db's simultaneously

Thanks for the help!

**Comment by includeamin** [comment]
Hi there,
did you check the [Docs](https://fastapi.tiangolo.com/tutorial/sql-databases/)

**Comment by YuriiMotov** [comment, marked as answer]
It should be something like this:

```python
@asynccontextmanager
async def lifespan(app: FastAPI):
    async with (
        Database(DATABASE_URL_1) as database_1,
        Database(DATABASE_URL_2) as database_2,
    ):
        yield {
            "database_1": database_1,
            "database_2": database_2,
        }


async def get_database_1():
    return app.state.database_1


async def get_database_2():
    return app.state.database_2


app = FastAPI()


@app.get("/users/1")
async def get_user(
    database_1=Depends(get_database_1),
    database_2=Depends(get_database_2),
):
    pass

```
